app/backend/src/tests/test_admin.py

```diff
@@ -0,0 +1,167 @@
+            with _session(_get_super_token()) as api:
+                res = api.GetUserEmailById(admin_pb2.GetUserEmailByIdRequest(user_id=normal_user.id))
+
+            assert res.email == NORMAL_USER_EMAIL
+            assert res.user_id == normal_user.id
+
+    @staticmethod
+    def test_GetEmailByUserName(testing_admin_api):
+        with session_scope() as session:
+            normal_user = _get_normal_user(session)
+            with _session(_get_super_token()) as api:
+                res = api.GetUserEmailByUserName(admin_pb2.GetUserEmailByUserNameRequest(username=normal_user.username))
+                assert res.email == NORMAL_USER_EMAIL
+                assert res.user_id == normal_user.id
+
+    @staticmethod
+    def test_GetBlockUser(testing_admin_api):
+        with session_scope() as session:
+            with _session(_get_super_token()) as api:
+                normal_user = _get_normal_user(session)
+                api.BlockUser(admin_pb2.BlockUserRequest(user_id=normal_user.id))
+                session.refresh(normal_user)
+                assert normal_user.is_banned
+
+    @staticmethod
+    def test_GetDeleteUser(testing_admin_api):
+        with session_scope() as session:
+            with _session(_get_super_token()) as api:
+                normal_user = _get_normal_user(session)
+                api.DeleteUser(admin_pb2.DeleteUserRequest(user_id=normal_user.id))
+                session.refresh(normal_user)
+                assert normal_user.is_deleted
+
+    @staticmethod
+    def test_CreateCommunityInvalidGeoJson(testing_admin_api):
+        with _session(_get_super_token()) as api:
+            with pytest.raises(grpc.RpcError) as e:
+                api.CreateCommunity(
+                    admin_pb2.CreateCommunityReq(
+                        name=COMMUNITY_NAME,
+                        slug=COMMUNITY_SLUG,
+                        description=COMMUNITY_DESCRIPTION,
+                        admin_ids=[],
+                        geojson=INVALID_GEOJSON,
+                    )
+                )
+            assert e.value.code() == grpc.StatusCode.INVALID_ARGUMENT
+            assert e.value.details() == errors.INVALID_MULTIPOLYGON
+
+    @staticmethod
+    def test_CreateCommunity(testing_admin_api):
+        with session_scope() as session:
+            with _session(_get_super_token()) as api:
+                api.CreateCommunity(
+                    admin_pb2.CreateCommunityReq(
+                        name=COMMUNITY_NAME,
+                        slug=COMMUNITY_SLUG,
+                        description=COMMUNITY_DESCRIPTION,
+                        admin_ids=[],
+                        geojson=VALID_GEOJSON_MULTIPOLYGON,
+                    )
+                )
+                community = session.query(Cluster).filter(Cluster.name == COMMUNITY_NAME).one()
+                assert community.description == COMMUNITY_DESCRIPTION
+                assert community.slug == COMMUNITY_SLUG
```

Comment:
Your testing boilerplate is quite different to the rest of the testing boilerplate. I think it'd be good to stay consistent with the rest of the tests.

Reply:
I don't think it is that structurally different besides replacing "magic"/duplicated strings with constants (which I think is generally a good practice in testing) and using more semantically precise methods to get the tokens needed for the tests we are executing here.

If there are reasons that these things are not good pratices I am happy to change them, otherwise I would not like introducing more "bad" practices because of consistency.

Reply:
I just mean that normally we have the tests as free functions that do the setup and teardown all in one test. The exception is the community tests that have their own setup etc, but that's because it's quite involved and time consuming so we wanted to reuse the same setup across tests.

Reply:
Ah, that makes sense, I copied the stetup from the communities tests. Will change it to the setup in one of the others.